### PR TITLE
Add loguru for logging

### PR DESCRIPTION
**Pull request by raghu017**
- Add loguru for logging
- Change PII logging to debug only

### CODE REVIEW

**Comment by vicondoa** [comment]
Hi @isafulf, this is part of the improvements we are making for Azure Open AI. This project already had a dependency on loguru, so @raghu017 is migrating all of the print statements to it along with switching all PII to debug statements. Let us know if there is anything you need from us.

**Comment by raghu017** [comment]
@isafulf Can you take a look at the PR?

File: datastore/providers/analyticdb_datastore.py [hunk 1]
```diff
@@ -252,7 +253,7 @@ def create_results(data):
                             QueryResult(query=query.query, results=results)
                         )
                 except Exception as e:
-                    print("error:", e)
+                    logger.exception("error:", e)
```
Comment (isafulf) [inline datastore/providers/analyticdb_datastore.py:256]: Doesn't logger.exception already logs the stack trace along with the error message?
So: `logger.exception("<descriptive message>")` should be sufficient, this applies to all cases in the PR

**Review by isafulf — COMMENTED** [review]
Thanks for doing this! Have you tested this yourself?

File: datastore/providers/analyticdb_datastore.py (hunk 1 above)
Comment (raghu017) [reply datastore/providers/analyticdb_datastore.py]: Pushed new changes, replaced exception with error. Exception is verbose stack trace and it is logging PII. Yes this is tested.

**Review by isafulf — APPROVED** [review]
LGTM thanks!

**Pull request merged**